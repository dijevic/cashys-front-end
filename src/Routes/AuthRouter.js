import React from 'react'
import { Route, Routes } from 'react-router-dom'
import { ResetPasswordComponent } from '../components/auth/resetpassword/ResetPassword'
import { Login } from '../pages/Login'
import { NotFound } from '../pages/NotFound'
import { SignUp } from '../pages/SignUp'

export const AuthRouter = () => {
    return (
        <Routes>

            <Route element={<Login />} path="/" />
            <Route element={<SignUp />} path="/signup" />
            <Route element={<ResetPasswordComponent />} path="/resetpassword" />
            <Route element={<NotFound />} path="*" />


        </Routes>
    )
}
