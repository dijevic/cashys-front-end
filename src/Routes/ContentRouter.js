import React from 'react'
import { Route, Routes } from 'react-router-dom'
import { Dashboard } from '../pages/Dashboard'
import { Profile } from '../pages/Profile'
import { NotFound } from '../pages/NotFound'


export const ContentRouter = React.memo(

    () => {



        return (
            <>

                <Routes>
                    <Route element={<Dashboard />} path="/" />
                    <Route element={<Profile />} path="/profile" />
                    <Route element={<NotFound />} path="*" />


                </Routes>
            </>

        )
    }

)