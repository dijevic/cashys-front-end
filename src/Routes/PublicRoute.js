import React from 'react'
import { Navigate } from 'react-router-dom'

export const PublicRoute = ({ isAuth, children: Children, ...rest }) => {

    return (!isAuth) ? <Children /> : <Navigate to="/dashboard/" />

}


