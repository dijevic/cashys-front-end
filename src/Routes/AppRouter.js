// thirt party importations

import React, { useEffect, useState } from 'react'
import {
    BrowserRouter as Router,
    Route,
    Routes,



} from 'react-router-dom'
import { Container, MainBackground } from '../components/coomon/div'
import { Spinner } from '../components/coomon/spinner'
import { NavBar } from '../components/ui/navbar/NavBar'
import { Footer } from '../pages/Footer'
import { NotFound } from '../pages/NotFound'
import { RenewTokenService } from '../services/renewTokenService'
import { useUserStore } from '../store/store'
import { AuthRouter } from './AuthRouter'
import { ContentRouter } from './ContentRouter'


// own imports
import { PrivateRoute } from './PrivateRoute'
import { PublicRoute } from './PublicRoute'


export const AppRouter = () => {



    const user = useUserStore(state => state.user)
    const setUser = useUserStore(state => state.setUser)
    const [checking, setChecking] = useState(true)
    const token = localStorage.getItem('token')

    useEffect(async () => {
        if (token) {
            RenewTokenService(setUser, setChecking)
        } else {
            setChecking(false)
        }
    }, [token])

    if (checking) {
        return (<Spinner />)
    }

    return (

        <Router>

            <>
                <NavBar />
                <MainBackground>
                    <Container>
                        <Routes>

                            <Route
                                element={
                                    <PublicRoute
                                        isAuth={user}
                                        children={AuthRouter} />}
                                path="/*" />

                            <Route
                                element={
                                    <PrivateRoute
                                        isAuth={user}
                                        children={ContentRouter} />}
                                path="/app/*" />


                            <Route path="*" element={<NotFound />} />
                        </Routes>
                    </Container>


                </MainBackground>

                <Footer />
            </>
        </Router>

    )
}


