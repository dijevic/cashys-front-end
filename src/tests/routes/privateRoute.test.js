import React from 'react'
import { shallow, ShallowWrapper } from "enzyme/build"

import { PrivateRoute } from "../../Routes/PrivateRoute"

describe('test for private route', () => {

    const children = <h1>this is a test child</h1>
    const wrapper = shallow(<PrivateRoute isAuth children={children} />)
    test('should match the snapshot correctly', () => {

        expect(wrapper).toMatchSnapshot()

    })


})