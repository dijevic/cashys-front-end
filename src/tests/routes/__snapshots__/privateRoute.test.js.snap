// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for private route should match the snapshot correctly 1`] = `<[object Object] />`;
